### PR TITLE
Mip7 annotations

### PR DESCRIPTION
**Pull request by moonso**
- [x] #1215 (SV cgh frequencies)
- [x] #1186 (gnomad-SV)
- [x] #1068 (local SV frequencies)
- [x] #868 (GenomicSuperDups)

### CODE REVIEW

**Comment by northwestwitch** [comment]
@henrikstranneheim could you write an example of a vcf line with GenomicSuperDups annotation? I know it's a string of items separated by '&',  but what type of items?

**Comment by northwestwitch** [comment]
@moonso I finished this PR. Could you check that it's fine? I've added tests for the new keys you've introduces before summer and I've added the GenomicSuperDups. These latter should be checked with care because I (kind of) have no idea what I'm doing with the transcript parsing here :)

File: scout/parse/variant/frequency.py [hunk 1]
```diff
@@ -20,7 +20,8 @@ def parse_frequencies(variant, transcripts):
     exac_keys = ['EXACAF']
     exac_max_keys = ['ExAC_MAX_AF', 'EXAC_MAX_AF']
 
-    gnomad_keys = ['GNOMADAF', 'GNOMAD_AF']
+    # Gnomad have both snv and sv frequencies
+    gnomad_keys = ['GNOMADAF', 'GNOMAD_AF', 'gnomad_svAF']
```
Comment (moonso) [inline scout/parse/variant/frequency.py]: Could you check that this is being displayed in the SV variants view?

**Comment by moonso** [comment]
And since this is originally my PR I can't approve... @northwestwitch or @dnil you have to do a review

**Comment by dnil** [comment]
I could have a look tomorrow, but feel free if you get there before!

File: scout/parse/variant/frequency.py (hunk 1 above)
Comment (northwestwitch) [reply scout/parse/variant/frequency.py]: I haven't tried with such a VCF. We could maybe pull the branch on stage and see if uploads a test case with these keys.

Comment (northwestwitch) [reply scout/parse/variant/frequency.py]: Is anyone using hasta stage?

**Comment by northwestwitch** [comment]
I've managed to re-upload SNV and SV variants for stage case 
https://scout-stage.scilifelab.se/cust002/15026-validation using this branch.  I don't know what is wrong with stage but it takes forever to open a variant. I'll try to check later

**Comment by northwestwitch** [comment]
SV are much faster to load. clingen mip frequency shows:
image

**Comment by moonso** [comment]
@northwestwitch I wonder if it get's slow since you make one request to entrez for each transcript with a refseq id every time a variant page is loaded. In general I think we should avoid making requests in the pages that one expects to load fast, we will be dependent on how fast there is a reply or if the service is down etc. Perhaps we can find another solution to how the transcript versions are added, like update ALL transcripts at some point. Or you only fetch this information when producing the ClinVar submission at this point.

**Comment by northwestwitch** [comment]
> I wonder if it get's slow since you make one request to entrez for each transcript with a refseq id every time a variant page is loaded.

I've been thinking about it. I'll try to modify this PR commenting the entrez request so we can see if that was the issue!

**Comment by northwestwitch** [comment]
Yes, that was the problem. I've fixed it in this PR in a way that refseq versions are collected only when calling the clinvar page. And now the variants are loaded fast on stage! But the PR went down in coverage unfortunately. Adding a test now!

**Comment by northwestwitch** [comment]
Example of a SV variant with updated annotations: 
https://scout-stage.scilifelab.se/cust002/15026-validation/sv/variants/e9a550af102743993ebe851d9189b89d
image

Checked in the database and unfortunately none of the 355 SNV variants loaded for this case has the superdups_fracmatches. (queried this way: {case_id: 'cuddlyoryx','genes.transcripts': {$exists: true}} )

@henrikstranneheim shall I try loading other cases among these: 


And hope to get one with the superdups?

**Comment by northwestwitch** [comment]
I've tried to load to scout all those test cases from the screenshot above but checking in the database I couldn't find any with the superdups_fracmatches in the transcript info. I don't know if it's because this PR doesn't work and the key is not captured or because the cases don't actually have variants in the tandem repeats. What should I do, should we trust the unittest (tests/parse/test_parse_transcripts.py) and that's it?

**Comment by moonso** [comment]
Have you tested to grep on the vcfs to see if they have superdups? We might need some help from @henrikstranneheim here

**Comment by northwestwitch** [comment]
> Have you tested to grep on the vcfs to see if they have superdups? We might need some help from @henrikstranneheim here

They have the info field at least but variants in the dups, I don't know

**Comment by henrikstranneheim** [comment]
It looks like this from case justhuskey:
```
1       10492   rs55998931      C       T ... INFO => T|downstream_gene_variant|MODIFIER|WASH7P|ENSG00000227232|Transcript|ENST00000488147|unprocessed_pseudogene|||||||||||3912|-1||HGNC|38034||||||||||Ensembl|C|C||||||||||||||||-11.244|2.005|-8.082|-9.238|0.435|-3.175|-2.739|-2.739||||0.982898&0.981582&0.992904&0.987856&0.986907&0.994232
```
GenomSuperDups here is: **0.982898&0.981582&0.992904&0.987856&0.986907&0.994232**
They are there but usually in upstream, downstream or in regulatory region or pseudogenes

**Comment by northwestwitch** [comment]
Still working on this. I'm having problem loading the gene region of WASH7P into scout:

> scout load region --case-id justhusky --hgnc-id 38034

image

**Comment by henrikstranneheim** [comment]
Looks like the index file is missing

**Comment by northwestwitch** [comment]
Index is there so I don't know why I get the error.

I've tried another approach, loading research variants. If I grep WASH7P in the research variants file (/home/proj/long-term-stage/MIP7.1-dev/housekeeper-bundles/justhusky/2019-08-23/justhusky_gatkcomb_rhocall_vt_frqf_cadd_vep_parsed_snpeff_ranked.vcf.gz) I see that the transcripts do have the superdups style -> float&float&etc

But  the if I open scout these variants are not there, I think because they're below the rank threshold, that I see in the code is 8 :(

**Comment by northwestwitch** [comment]
Ah wait, the index is there just for snvs and svs, not the strs..

**Comment by northwestwitch** [comment]
Yess, I managed to upload research variants in that gene using this command: 
```
scout load variants --hgnc-symbol WASH7P justhusky --snv-research --rank-treshold -20 --force
```
And the variant mentioned by @henrikstranneheim is in the database but It doesn't show superdups:
https://scout-stage.scilifelab.se/cust000/SWEDAC-2019-HG-05-Trio-NovaSeq/fa342e72621f6dd522d71c93c469d26f

So there's an error somewhere in the code. I'll fix it!

**Comment by northwestwitch** [comment]
Mappability is collected from annotations and showed on variants page now: 
image

As soon as travis finished I think I'll approve and merge

**Comment by dnil** [comment]
Nice! Could we figure out how many digits are significant and round accordingly?

**Comment by northwestwitch** [comment]
> Nice! Could we figure out how many digits are significant and round accordingly?

The score is the fraction of matching bases, so what about 3 numbers after comma ?

**Comment by dnil** [comment]
Sounds about right; they are all >1000 bp at least. In actual practice, having the seg_dup feature show at all on a variant will constitute a warning flag.

**Comment by northwestwitch** [comment]
There's another issue I see. Entrez e-utilities are down so the refseq version test I wrote for the previous PR is failing

**Comment by dnil** [comment]
😞 Ok. So, that would again speak strongly for seeking a solution where we dl all of the transcript versions at once, wouldn't it?

**Comment by northwestwitch** [comment]
> Ok. So, that would again speak strongly for seeking a solution where we dl all of the transcript versions at once, wouldn't it?

That would be the best solution. Yes

**Comment by northwestwitch** [comment]
All good. Entrez is working and the test passes it looks. This is the final looks of the superdups:
image

@dnil do I merge?

**Comment by northwestwitch** [comment]
NML has just confirmed that the eutilities were down yesterday, so we don't require an API key as I was starting to suspect:

```
Dear Colleague,

The E-Utilities were experiencing server outages for a period this morning.  The issue should be resolved.  Please let us know if you experience additional errors.

Apologies for the inconvenience.

Best regards,

User Services
NCBI | NLM | NIH
```

I'm going to merge now!

**Comment by dnil** [comment]
Good! The transcript number part is stuff for another issue/PR anyway, eventually, so don't worry about that here. Seg_dup is now fine. 

What is ClinGen MIP? One of the research cohorts, captured with SVDB, or  loqusDB? At any rate, we need to document the f sources, and if possible add a cohort size to each. Also time for a fred hover, but both of these are for another PR as well.. 😅

**Comment by henrikstranneheim** [comment]
What is ClinGen MIP? 
It is the one of svdb merge of roughly 2000 WGS SV (grch37).

**Pull request merged**